### ISSUE TITLE
Sendable() returns false when communicating with a particular peers implementation of amqp.

### ISSUE THREAD

**Issue by anthonyvercolano** [amqp/rhea#210]
I have some code that is sending to a peer.  Before we actually call rhea send we first check if the link is sendable().

This code is used to send to peer A and to peer B.  NOT at the same time, not in the same process, or thread.

When sending to peer A the calls to sendable() are always returning true.  Yay!

When sending to peer B the call to sendable() will return false after doing 2048 sends

The code below is from session.js.

CircularBuffer.prototype.available = function () {
    return this.capacity - this.size;
};

At the point of the 2048th send with peer B this.size == 2048 and this.capacity is 2048.
At the point of the 2048th send with peer A this.capacity is 0.

Peer B is not using the same amqp implementation as peer A.

logs for sending to peer A and to peer B both show that the dispositions are received for each and every send until we send the 2048th to B cause we don't give any new sends to dispose.

This is a typical disposition for a send to peer B:
rhea:frames [connection-1]:0 <- disposition#15 {"role":true,"first":28,"settled":true,"state":[]}

Would you have any opinion on why the sends to peer B never seem to decrement this.free?

**Comment by grs** [comment]
No, that looks odd, possibly a bug. I asume you are using auto-settle in both cases and the same sender settlement mode?

Outgoing.on_disposition (in session.js) is where the disposition gets handled. That should then result in a settled event being generated during Outgoing.process, and the auto-settle will then mark it locally settled. Once remotely and locally settled it should be removed, again in Outgoing.processed.

**Comment by anthonyvercolano** [comment]
I believe we do not use auto-settle on the sender link.  Link options are identical on both peers.

Here is the local attach for both peers:
rhea:frames [connection-1]:0 -> attach#12 {"name":"rheaSender_8a4cd3e1-ce8f-4c79-b2b4-ab59db0e136c","handle":3,"source":[],"target":["/devices/xxxx/messages/events"],"properties":{"com.microsoft:client-version":"azure-iot-device/1.9.4 (node v10.15.3; Ubuntu Linux 18.04; x64)"}} 

rhea:frames [connection-1]:0 -> attach#12 {"name":"rheaSender_31ec900e-2913-4ff5-a00a-3f410a18524d","handle":3,"source":[],"target":["/devices/yyyy/messages/events"],"properties":{"com.microsoft:client-version":"azure-iot-device/1.9.4 (node v10.15.3; Ubuntu Linux 18.04; x64)"}}

Also, for these particular sending links, the delivery object returned from the rhea send has .settled == false.

**Comment by grs** [comment]
The settled flag will be false until locally set. By default an auto-settle handler is registered that will settle locally whenever the peer settles (i.e. on getting a disposition). If that is turned off then the application would have to set settled=true itself, e.g. in response to the settled event.

The disposition looks fine (do you have an example from the other peer that works, just for comparison?), so either there is a bug in handling that, or perhaps the local settled flag is not being set. Once both settled and remote_settled are true, the delivery should become eligible to be removed.

**Comment by anthonyvercolano** [comment]
I'm having difficulty copy / pasting between the vm and the browser at this point.  I assure you they look exactly the same.

As for marking it settled locally.  The function callback for the sender accepted event has a pointer to the delivery object in it's context.  Are you suggesting marking the settled property of that delivery object as true?

**Comment by grs** [comment]
> The function callback for the sender accepted event has a pointer to the delivery object in it's context. Are you suggesting marking the settled property of that delivery object as true?

You could try it, but if the only thing different between the two cases is the peer (not the local code) then I don't think that can be the issue. Can you add some debug logging or something and log the processing of deliveries in Outgoing.process, to see if settled and remote_settled are as expected.
